novice-goals.Rmd

```diff
@@ -0,0 +1,210 @@
+# Outline {#outline}
+
+This outline describes the questions that the novice courses on R and Python will answer.
+The advanced course can then assume that learners have hands-on experience with these topics
+but nothing more.
+
+## Data manipulation
+
+-   How can I read tabular data into a program?
+    -   CSV
+    -   reading files
+-   How can I select subsets of my data?
+    -   select
+    -   filter
+    -   Boolean conditions
+-   How can I calculate new values?
+    -   mutate
+    -   ifelse
+-   How can I tell what's gone wrong in my programs?
+    -   reading error messages
+    -   the difference between syntax and runtime errors
+-   How can I operate on subsets of my data?
+    -   group
+    -   summarize
+-   How can I work with two or more datasets?
+    -   join
+-   What *isn't* included?
+    -   anything other than reasonably tidy tabular data
+    -   map
+    -   loops and conditionals
+
+## Programming
+
+-   What are the different ways I can interact with software?
+    -   console
+    -   scripts
+    -   libraries/packages?
```

Comment:
maybe mention libraries as a reusable extension of scripts/functions

Reply:
I don't think we can do that yet - they haven't seen how to write functions of their own.